File: src/main/java/com/box/BoxTester.java
package com.box;

import java.util.Scanner;

public class BoxTester {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Please enter object's length:");
        float length = scanner.nextFloat();
        System.out.print("Please enter object's width:");
        float width = scanner.nextFloat();
        System.out.print("Please enter object's height:");
        int height = scanner.nextInt();
        if (Box3.getInstance().validate(length, width, height)) {
            System.out.println("Box3");
        } else if (Box5.getInstance().validate(length, width, height)) {
            System.out.println("Box5");
        } else {
            System.out.println("使用XBox360!");
        }
    }
}

File: src/main/kotlin/com/box/BoxTester.kt
package com.trainKt

fun main(args: Array<String>) {
    print("Please enter object's length:")
    val lenght = readLine()!!.toFloat()
    print("Please enter object's width:")
    val width = readLine()!!.toFloat()
    print("Please enter object's height:")
    val height = readLine()!!.toInt()
    val b = Box.selectBox(lenght, width, height)
    print(b)
}

abstract class Box(val length:Float, val width:Float, val height:Int) {
    companion object {
        val boxs = mutableListOf(Box3(), Box5())
        fun selectBox(length:Float, width:Float, height:Int ): String{
            for (box in boxs) {
                if (box.validate(length, width, height)) {
                    return "${box.name()}, 長：${box.length}, 寬：${box.width}, 高：${box.height}"
                }
            }
            return "暫時沒有符合的箱子"
        }
    }
    fun validate(length:Float, width:Float, height:Int): Boolean {
        return this.length >= length && this.width >= width && this.height >= height
    }
    abstract fun name() : String
}

class Box3: Box(23f, 14f,13) {
    override fun name(): String {
        return "Box3"
    }
}

class Box5 : Box(39.5f, 27.5f, 23) {
    override fun name(): String {
        return "Box5"
    }

}

